src/mscorlib/src/System/IO/Stream.cs

```diff
@@ -900,6 +900,13 @@ private sealed class NullStream : Stream
                 get { return 0; }
                 set {}
             }
+            
+            public override Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken)
+            {
+                if (cancellationToken.IsCancellationRequested)
```

Comment:
> but Stream.Null already overrides ReadAsync/WriteAsync for perf, so why not CopyToAsync as well?

If it didn't override ReadAsync/WriteAsync, the base implementations would end up allocating.  That's not the case with CopyToAsync, where the base implementation will call ReadAsync and return immediately, without allocating or incurring much cost at all.